tests/1d_flow.cc

```diff
@@ -0,0 +1,238 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the adaflo authors
+//
+// This file is part of the adaflo library.
+//
+// The adaflo library is free software; you can use it, redistribute it,
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.  The full text of the
+// license can be found in the file LICENSE at the top level of the adaflo
+// distribution.
+//
+// --------------------------------------------------------------------------
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/function_time.h>
+#include <deal.II/base/index_set.h>
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/utilities.h>
+
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_in.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/manifold_lib.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/la_parallel_vector.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <adaflo/navier_stokes.h>
+#include <adaflo/time_stepping.h>
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+
+using namespace dealii;
+
+/**
+ * Simple test case for 1d Navier-Stokes
+ *
+ *
+ * velocity(t=0) = 2
+ *                             L=2.5
+ * (pressure=2) +-------------------------------+ (p=1)
+ *              |--> x
+ *
+ */
+
+
+template <int dim>
+class InflowVelocity : public Function<dim>
+{
+public:
+  InflowVelocity()
+    : Function<dim>()
+  {}
+
+  double
+  value(const Point<dim> &p, const unsigned int component = 0) const override
+  {
+    (void)p;
+    (void)component;
+    if constexpr (dim == 1)
```

Comment:
Do we need the `constexpr` here? I don't see it immediately, so why not simply use a regular `if`? (Efficiency is not affected because the condition is known at compile time.)

Reply:
Good point, it is not needed here. I replaced this function by a `dealii::Functions::ConstantFunction` since I apply only a constant velocity field. This is actually a leftover from when I was playing around with the velocity field.